google/google-how-to/gce/set-hostname-using-cloudinit.rst

```diff
@@ -1,20 +1,29 @@
 Set hostname of GCE instances
-=================================
+=============================
 
-The hostname of GCE instances can be set using multiple methods. Apart from setting it manually, one option is to use a set-up tool like cloud-init, while Google's preferred method is to use its DHCP service. Both these options are considered here.
+The hostname of GCE instances can be set using multiple methods. Google's preferred method is to use its DHCP service, which requires you to choose a fully qualified DNS name (FQDN), e.g. something like test123.test.com. If you don't want an FQDN or if you want a specific format for consistent naming across clouds, you can use a set-up tool like cloud-init to set your hostname.
 
-.. _use_cloud-init:
+Both these methods are described here. Also, due to a recent hostname-related update on GCP, you might have to make some additional changes for GCE images that use Ubuntu 24.04 and later. These are explained at the end.
 
-Option 1: Use cloud-init
-------------------------
 
-You can set the required hostname by programming the cloud-init metadata. For this, use the `gcloud compute instances add-metadata`_ command:
+Using DHCP (Google's preferred method)
+--------------------------------------
+
+By default, Google's DHCP service sets the hostname to an automatically generated internal DNS name. 
+
+To set your own custom name, follow the instructions given in `Create a VM instance with a custom hostname`_. In this case, the DHCP service will additionally provide the custom name and will prioritise it to be the default hostname. However, as mentioned earlier, the custom name needs to be an FQDN and you also need to create a corresponding DNS record for it.
+
+
+Using cloud-init
+----------------
+
+cloud-init uses the ``hostname`` command to programmatically set the hostname. You just need to configure its metadata with the required hostname and use the `gcloud compute instances add-metadata`_ command:
```

Comment:
No "just" in the second sentence, I think.

Usually for cloud-init, people use `metadata-from-file` rather than `add-metadata` so I think we should use that as the example command here.

Reply:
Removed 'just'.

Isn't ``metadata-from-file`` a parameter to be set in commands such as ``add-metadata`` and ``create``? The command we have shown includes it as a parameter.

Reply:
Oh I didn't even realize that `add-metadata` was a peer command to `create` for some reason I thought it was a subcommand like `metadata-from-file`.  I have only ever used `gcloud compute instances create` but I think your wording is good since we show the `create` usage immediately after.